velox/docs/functions/presto/decimal.rst

```diff
@@ -0,0 +1,201 @@
+=================
+Decimal Operators
+=================
+
+DECIMAL type is designed to represent floating point numbers precisely.
+Mathematical operations on decimal values are exact, except for division. On
+the other hand, DOUBLE and REAL types are designed to represent floating point
+numbers approximately. Mathematical operations on double and real values are
+approximate.
+
+For example, the number 5,000,000,000,000,000 can be represented using DOUBLE.
+However, the number 5,000,000,000,000,000.15 cannot be represented using
+DOUBLE, but it can be represented using DECIMAL. See
+https://en.wikipedia.org/wiki/Double-precision_floating-point_format for more
+details.
+
+DECIMAL type has 2 parameters: precision and scale. Precision is the total
+number of digits used to represent the number. Scale is the number of digits
+after the decimal point. Naturally, scale must not exceed precision. In
+addition, precision cannot exceed 38.
+
+::
+
+	decimal(p, s)
+
+	p >= 1 && p <= 38
+	s >= 0 && s <= p
+
+The number 5,000,000,000,000,000.15 can be represented using DECIMAL(18, 2).
+This number needs at least 18 digits (precision) of which at least 2 digits
+must appear after the decimal point (scale). This number can be represented
+using any DECIMAL type where scale >= 2 and precision is >= scale + 16.
+
+Addition and Subtraction
+------------------------
+
+To represent the results of adding two decimal numbers we need to use max
+(s1, s2) digits after the decimal point and max(p1 - s1, p2 - s2) + 1 digits
+before the decimal point.
+
+::
+
+	p = max(p1 - s1, p2 - s2) + 1 + max(s1, s2)
+	s = max(s1, s2)
+
+It is easiest to understand this formula by thinking about column addition where
+we place two numbers one under the other and line up decimal points.
+
+::
+
+        1.001
```

Comment:
This block's formatting is off in the file view for some reason.
https://github.com/facebookincubator/velox/blob/a8f2c40ad339568483306f394ee5111c83b0196e/velox/docs/functions/presto/decimal.rst

Reply:
It seems to look good in the generated docs:

<img width="600" alt="Screenshot 2024-04-02 at 12 52 08 PM" src="https://github.com/facebookincubator/velox/assets/27965151/567fc6f0-a14d-4860-84ec-5dd7ab8570c9">